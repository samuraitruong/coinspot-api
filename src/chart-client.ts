import qs from 'qs';
import { BaseClient } from './base-client';
import { HistoryRequest } from './types';
export class ChartClient extends BaseClient {
  
  constructor() {
    super("", "", "https://www.coinspot.com.au/charts");
  }

   async getHistory(request: HistoryRequest) {
    const url =  "history_basic?" + qs.stringify(request);
    return this.getRequest(url)
  }
}
