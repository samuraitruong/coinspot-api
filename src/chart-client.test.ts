import { ChartClient } from "./chart-client";
describe("ChartClient tests", () => {
  const client = new ChartClient();
  it("getHistory test", async () => {
    const zil = await client.getHistory({ symbol: "ZIL" });
    expect(zil.length).toBeGreaterThan(0);
  });
});
