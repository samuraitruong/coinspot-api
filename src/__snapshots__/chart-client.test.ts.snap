// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartClient tests getHistory test 1`] = `
Array [
  Array [
    1523059200000,
    0.05007365,
  ],
  Array [
    1523318400000,
    0.05202015,
  ],
  Array [
    1523577600000,
    0.07793847,
  ],
  Array [
    1523836800000,
    0.08423992,
  ],
  Array [
    1524096000000,
    0.08680758,
  ],
  Array [
    1524355200000,
    0.1181544,
  ],
  Array [
    1524614400000,
    0.13099973,
  ],
  Array [
    1524873600000,
    0.12099312,
  ],
  Array [
    1525132800000,
    0.13720572,
  ],
  Array [
    1525392000000,
    0.17427677,
  ],
  Array [
    1525651200000,
    0.18607817,
  ],
  Array [
    1525910400000,
    0.27050545,
  ],
  Array [
    1526169600000,
    0.2443326,
  ],
  Array [
    1526428800000,
    0.20843122,
  ],
  Array [
    1526688000000,
    0.18665042,
  ],
  Array [
    1526947200000,
    0.19034526,
  ],
  Array [
    1527206400000,
    0.16723093,
  ],
  Array [
    1527465600000,
    0.15598613,
  ],
  Array [
    1527724800000,
    0.14768761,
  ],
  Array [
    1527984000000,
    0.15997944,
  ],
  Array [
    1528243200000,
    0.16829091,
  ],
  Array [
    1528502400000,
    0.16829657,
  ],
  Array [
    1528761600000,
    0.13834225,
  ],
  Array [
    1529020800000,
    0.12699001,
  ],
  Array [
    1529280000000,
    0.11524392,
  ],
  Array [
    1529539200000,
    0.12181087,
  ],
  Array [
    1529798400000,
    0.09348363,
  ],
  Array [
    1530057600000,
    0.08582727,
  ],
  Array [
    1530316800000,
    0.09113899,
  ],
  Array [
    1530576000000,
    0.10440255,
  ],
  Array [
    1530835200000,
    0.11677077,
  ],
  Array [
    1531094400000,
    0.11564456,
  ],
  Array [
    1531353600000,
    0.09543181,
  ],
  Array [
    1531612800000,
    0.09150674,
  ],
  Array [
    1531872000000,
    0.11493425,
  ],
  Array [
    1532131200000,
    0.09682724,
  ],
  Array [
    1532390400000,
    0.09548223,
  ],
  Array [
    1532649600000,
    0.10060221,
  ],
  Array [
    1532908800000,
    0.09943676,
  ],
  Array [
    1533168000000,
    0.0841981,
  ],
  Array [
    1533427200000,
    0.07366471,
  ],
  Array [
    1533686400000,
    0.06284355,
  ],
  Array [
    1533945600000,
    0.05299725,
  ],
  Array [
    1534204800000,
    0.04147995,
  ],
  Array [
    1534464000000,
    0.04183829,
  ],
  Array [
    1534723200000,
    0.05309132,
  ],
  Array [
    1534982400000,
    0.0473112,
  ],
  Array [
    1535241600000,
    0.05658515,
  ],
  Array [
    1535500800000,
    0.06561213,
  ],
  Array [
    1535760000000,
    0.0616161,
  ],
  Array [
    1536019200000,
    0.06210794,
  ],
  Array [
    1536278400000,
    0.05303329,
  ],
  Array [
    1536537600000,
    0.04855595,
  ],
  Array [
    1536796800000,
    0.04543317,
  ],
  Array [
    1537056000000,
    0.04752754,
  ],
  Array [
    1537315200000,
    0.04581694,
  ],
  Array [
    1537574400000,
    0.05165447,
  ],
  Array [
    1537833600000,
    0.04904293,
  ],
  Array [
    1538092800000,
    0.0494876,
  ],
  Array [
    1538352000000,
    0.04921607,
  ],
  Array [
    1538611200000,
    0.05277118,
  ],
  Array [
    1538870400000,
    0.05145711,
  ],
  Array [
    1539129600000,
    0.05358496,
  ],
  Array [
    1539388800000,
    0.04502462,
  ],
  Array [
    1539648000000,
    0.04729212,
  ],
  Array [
    1539907200000,
    0.04701418,
  ],
  Array [
    1540166400000,
    0.04742695,
  ],
  Array [
    1540425600000,
    0.0488464,
  ],
  Array [
    1540684800000,
    0.05012417,
  ],
  Array [
    1540944000000,
    0.04888941,
  ],
  Array [
    1541203200000,
    0.05039162,
  ],
  Array [
    1541462400000,
    0.0501099,
  ],
  Array [
    1541721600000,
    0.04899166,
  ],
  Array [
    1541980800000,
    0.04719761,
  ],
  Array [
    1542240000000,
    0.03956494,
  ],
  Array [
    1542499200000,
    0.03663086,
  ],
  Array [
    1542758400000,
    0.0241684,
  ],
  Array [
    1543017600000,
    0.02326243,
  ],
  Array [
    1543276800000,
    0.01939048,
  ],
  Array [
    1543536000000,
    0.02643106,
  ],
  Array [
    1543795200000,
    0.02461745,
  ],
  Array [
    1544054400000,
    0.02120445,
  ],
  Array [
    1544313600000,
    0.01919883,
  ],
  Array [
    1544572800000,
    0.01875529,
  ],
  Array [
    1544832000000,
    0.01782606,
  ],
  Array [
    1545091200000,
    0.02006565,
  ],
  Array [
    1545350400000,
    0.02457146,
  ],
  Array [
    1545609600000,
    0.02852337,
  ],
  Array [
    1545868800000,
    0.02648056,
  ],
  Array [
    1546128000000,
    0.02986565,
  ],
  Array [
    1546387200000,
    0.02968855,
  ],
  Array [
    1546646400000,
    0.02969238,
  ],
  Array [
    1546905600000,
    0.0317173,
  ],
  Array [
    1547164800000,
    0.02885677,
  ],
  Array [
    1547424000000,
    0.02569992,
  ],
  Array [
    1547683200000,
    0.03262328,
  ],
  Array [
    1547942400000,
    0.03159092,
  ],
  Array [
    1548201600000,
    0.03263246,
  ],
  Array [
    1548460800000,
    0.03123695,
  ],
  Array [
    1548720000000,
    0.02979732,
  ],
  Array [
    1548979200000,
    0.02567947,
  ],
  Array [
    1549238400000,
    0.02430746,
  ],
  Array [
    1549497600000,
    0.02332011,
  ],
  Array [
    1549756800000,
    0.02461851,
  ],
  Array [
    1550016000000,
    0.02430212,
  ],
  Array [
    1550275200000,
    0.0239955,
  ],
  Array [
    1550534400000,
    0.02552156,
  ],
  Array [
    1550793600000,
    0.02594312,
  ],
  Array [
    1551052800000,
    0.02411355,
  ],
  Array [
    1551312000000,
    0.02472722,
  ],
  Array [
    1551571200000,
    0.02447348,
  ],
  Array [
    1551830400000,
    0.02436579,
  ],
  Array [
    1552089600000,
    0.02476002,
  ],
  Array [
    1552348800000,
    0.02509343,
  ],
  Array [
    1552608000000,
    0.02622952,
  ],
  Array [
    1552867200000,
    0.02597352,
  ],
  Array [
    1553126400000,
    0.0277043,
  ],
  Array [
    1553385600000,
    0.02797999,
  ],
  Array [
    1553644800000,
    0.0264748,
  ],
  Array [
    1553904000000,
    0.02749365,
  ],
  Array [
    1554163200000,
    0.02941922,
  ],
  Array [
    1554422400000,
    0.03261746,
  ],
  Array [
    1554681600000,
    0.03564499,
  ],
  Array [
    1554940800000,
    0.03267057,
  ],
  Array [
    1555200000000,
    0.02898185,
  ],
  Array [
    1555459200000,
    0.03032036,
  ],
  Array [
    1555718400000,
    0.03170164,
  ],
  Array [
    1555977600000,
    0.02993827,
  ],
  Array [
    1556236800000,
    0.02516633,
  ],
  Array [
    1556496000000,
    0.02465576,
  ],
  Array [
    1556755200000,
    0.0252617,
  ],
  Array [
    1557014400000,
    0.02318565,
  ],
  Array [
    1557273600000,
    0.0225884,
  ],
  Array [
    1557532800000,
    0.02238867,
  ],
  Array [
    1557792000000,
    0.02423757,
  ],
  Array [
    1558051200000,
    0.02768398,
  ],
  Array [
    1558310400000,
    0.02628544,
  ],
  Array [
    1558569600000,
    0.03225735,
  ],
  Array [
    1558828800000,
    0.03028145,
  ],
  Array [
    1559088000000,
    0.03202302,
  ],
  Array [
    1559347200000,
    0.02990738,
  ],
  Array [
    1559606400000,
    0.03086745,
  ],
  Array [
    1559865600000,
    0.0340915,
  ],
  Array [
    1560124800000,
    0.03226315,
  ],
  Array [
    1560384000000,
    0.03425028,
  ],
  Array [
    1560643200000,
    0.03398793,
  ],
  Array [
    1560902400000,
    0.03379253,
  ],
  Array [
    1561161600000,
    0.02946414,
  ],
  Array [
    1561420800000,
    0.02969334,
  ],
  Array [
    1561680000000,
    0.02376854,
  ],
  Array [
    1561939200000,
    0.02403431,
  ],
  Array [
    1562198400000,
    0.02387025,
  ],
  Array [
    1562457600000,
    0.02464972,
  ],
  Array [
    1562716800000,
    0.02527572,
  ],
  Array [
    1562976000000,
    0.02145374,
  ],
  Array [
    1563235200000,
    0.01746521,
  ],
  Array [
    1563494400000,
    0.01616222,
  ],
  Array [
    1563753600000,
    0.01632313,
  ],
  Array [
    1564012800000,
    0.01526276,
  ],
  Array [
    1564272000000,
    0.01537053,
  ],
  Array [
    1564531200000,
    0.01597212,
  ],
  Array [
    1564790400000,
    0.0154681,
  ],
  Array [
    1565049600000,
    0.01509685,
  ],
  Array [
    1565308800000,
    0.01283659,
  ],
  Array [
    1565568000000,
    0.01352519,
  ],
  Array [
    1565827200000,
    0.01161377,
  ],
  Array [
    1566086400000,
    0.01166945,
  ],
  Array [
    1566345600000,
    0.01133925,
  ],
  Array [
    1566604800000,
    0.01164615,
  ],
  Array [
    1566864000000,
    0.0127994,
  ],
  Array [
    1567123200000,
    0.01075354,
  ],
  Array [
    1567382400000,
    0.01082409,
  ],
  Array [
    1567641600000,
    0.01067481,
  ],
  Array [
    1567900800000,
    0.0103691,
  ],
  Array [
    1568160000000,
    0.01025101,
  ],
  Array [
    1568419200000,
    0.01032671,
  ],
  Array [
    1568678400000,
    0.01053277,
  ],
  Array [
    1568937600000,
    0.01098181,
  ],
  Array [
    1569196800000,
    0.01013069,
  ],
  Array [
    1569456000000,
    0.00766861,
  ],
  Array [
    1569715200000,
    0.00784902,
  ],
  Array [
    1569974400000,
    0.00789024,
  ],
  Array [
    1570233600000,
    0.00815059,
  ],
  Array [
    1570492800000,
    0.00884389,
  ],
  Array [
    1570752000000,
    0.00881382,
  ],
  Array [
    1571011200000,
    0.00859449,
  ],
  Array [
    1571270400000,
    0.00791143,
  ],
  Array [
    1571529600000,
    0.0078429,
  ],
  Array [
    1571788800000,
    0.0077825,
  ],
  Array [
    1572048000000,
    0.00792338,
  ],
  Array [
    1572307200000,
    0.00859016,
  ],
  Array [
    1572566400000,
    0.00840699,
  ],
  Array [
    1572825600000,
    0.00900423,
  ],
  Array [
    1573084800000,
    0.00955874,
  ],
  Array [
    1573344000000,
    0.00904623,
  ],
  Array [
    1573603200000,
    0.00880147,
  ],
  Array [
    1573862400000,
    0.01014354,
  ],
  Array [
    1574121600000,
    0.0091984,
  ],
  Array [
    1574380800000,
    0.00788381,
  ],
  Array [
    1574640000000,
    0.00694917,
  ],
  Array [
    1574899200000,
    0.00780122,
  ],
  Array [
    1575158400000,
    0.00810194,
  ],
  Array [
    1575417600000,
    0.00859962,
  ],
  Array [
    1575676800000,
    0.00932001,
  ],
  Array [
    1575936000000,
    0.00910494,
  ],
  Array [
    1576195200000,
    0.00777858,
  ],
  Array [
    1576454400000,
    0.00766752,
  ],
  Array [
    1576713600000,
    0.00782424,
  ],
  Array [
    1576972800000,
    0.00730457,
  ],
  Array [
    1577232000000,
    0.0070755,
  ],
  Array [
    1577491200000,
    0.00701195,
  ],
  Array [
    1577750400000,
    0.00665079,
  ],
  Array [
    1578009600000,
    0.00650603,
  ],
  Array [
    1578268800000,
    0.00692557,
  ],
  Array [
    1578528000000,
    0.00706922,
  ],
  Array [
    1578787200000,
    0.00704811,
  ],
  Array [
    1579046400000,
    0.00762018,
  ],
  Array [
    1579305600000,
    0.00819418,
  ],
  Array [
    1579564800000,
    0.00759945,
  ],
  Array [
    1579824000000,
    0.00751473,
  ],
  Array [
    1580083200000,
    0.00770344,
  ],
  Array [
    1580342400000,
    0.00846494,
  ],
  Array [
    1580601600000,
    0.00930343,
  ],
  Array [
    1580860800000,
    0.01044236,
  ],
  Array [
    1581120000000,
    0.01272742,
  ],
  Array [
    1581379200000,
    0.01173861,
  ],
  Array [
    1581638400000,
    0.01240346,
  ],
  Array [
    1581897600000,
    0.01197207,
  ],
  Array [
    1582156800000,
    0.01055827,
  ],
  Array [
    1582416000000,
    0.01165195,
  ],
  Array [
    1582675200000,
    0.01024182,
  ],
  Array [
    1582934400000,
    0.00943574,
  ],
  Array [
    1583193600000,
    0.00987953,
  ],
  Array [
    1583452800000,
    0.00981188,
  ],
  Array [
    1583712000000,
    0.00892328,
  ],
  Array [
    1583971200000,
    0.00907602,
  ],
  Array [
    1584230400000,
    0.00582707,
  ],
  Array [
    1584489600000,
    0.00621977,
  ],
  Array [
    1584748800000,
    0.00696744,
  ],
  Array [
    1585008000000,
    0.00698665,
  ],
  Array [
    1585267200000,
    0.00678355,
  ],
  Array [
    1585526400000,
    0.00600992,
  ],
  Array [
    1585785600000,
    0.00666162,
  ],
  Array [
    1586044800000,
    0.00710031,
  ],
  Array [
    1586304000000,
    0.00771645,
  ],
  Array [
    1586563200000,
    0.00692141,
  ],
  Array [
    1586822400000,
    0.00674855,
  ],
  Array [
    1587081600000,
    0.00702801,
  ],
  Array [
    1587340800000,
    0.0070696,
  ],
  Array [
    1587600000000,
    0.00700661,
  ],
  Array [
    1587859200000,
    0.00778418,
  ],
  Array [
    1588118400000,
    0.00800724,
  ],
  Array [
    1588377600000,
    0.00963524,
  ],
  Array [
    1588636800000,
    0.01150936,
  ],
  Array [
    1588896000000,
    0.01139889,
  ],
  Array [
    1589155200000,
    0.01087901,
  ],
  Array [
    1589414400000,
    0.0152237,
  ],
  Array [
    1589673600000,
    0.0178371,
  ],
  Array [
    1589932800000,
    0.01549918,
  ],
  Array [
    1590192000000,
    0.01594711,
  ],
  Array [
    1590451200000,
    0.0197371,
  ],
  Array [
    1590710400000,
    0.01934268,
  ],
  Array [
    1590969600000,
    0.0228921,
  ],
  Array [
    1591228800000,
    0.02677215,
  ],
  Array [
    1591488000000,
    0.03170937,
  ],
  Array [
    1591747200000,
    0.03646006,
  ],
  Array [
    1592006400000,
    0.03902599,
  ],
  Array [
    1592265600000,
    0.03728093,
  ],
  Array [
    1592524800000,
    0.03127139,
  ],
  Array [
    1592784000000,
    0.02996426,
  ],
  Array [
    1593043200000,
    0.02945729,
  ],
  Array [
    1593302400000,
    0.02384878,
  ],
  Array [
    1593561600000,
    0.02592908,
  ],
  Array [
    1593820800000,
    0.02516781,
  ],
  Array [
    1594080000000,
    0.02828693,
  ],
  Array [
    1594339200000,
    0.02951636,
  ],
  Array [
    1594598400000,
    0.02891155,
  ],
  Array [
    1594857600000,
    0.02822089,
  ],
  Array [
    1595116800000,
    0.02649663,
  ],
  Array [
    1595376000000,
    0.02618311,
  ],
  Array [
    1595635200000,
    0.02700784,
  ],
  Array [
    1595894400000,
    0.02450329,
  ],
  Array [
    1596153600000,
    0.02630816,
  ],
  Array [
    1596412800000,
    0.02582652,
  ],
  Array [
    1596672000000,
    0.02833856,
  ],
  Array [
    1596931200000,
    0.03804453,
  ],
  Array [
    1597190400000,
    0.03756048,
  ],
  Array [
    1597449600000,
    0.0361424,
  ],
  Array [
    1597708800000,
    0.03634143,
  ],
  Array [
    1597968000000,
    0.03245653,
  ],
  Array [
    1598227200000,
    0.03088422,
  ],
  Array [
    1598486400000,
    0.03200808,
  ],
  Array [
    1598745600000,
    0.02924796,
  ],
  Array [
    1599004800000,
    0.03068075,
  ],
  Array [
    1599264000000,
    0.02364541,
  ],
  Array [
    1599523200000,
    0.0218266,
  ],
  Array [
    1599782400000,
    0.02459459,
  ],
  Array [
    1600041600000,
    0.02362742,
  ],
  Array [
    1600300800000,
    0.02391525,
  ],
  Array [
    1600560000000,
    0.0239069,
  ],
  Array [
    1600819200000,
    0.02040336,
  ],
  Array [
    1601078400000,
    0.02348029,
  ],
  Array [
    1601337600000,
    0.02201627,
  ],
  Array [
    1601596800000,
    0.0277676,
  ],
  Array [
    1601856000000,
    0.0306752,
  ],
  Array [
    1602115200000,
    0.02732573,
  ],
  Array [
    1602374400000,
    0.02821199,
  ],
  Array [
    1602633600000,
    0.02882303,
  ],
  Array [
    1602892800000,
    0.02630235,
  ],
  Array [
    1603152000000,
    0.02894963,
  ],
  Array [
    1603411200000,
    0.02831943,
  ],
  Array [
    1603670400000,
    0.02766021,
  ],
  Array [
    1603929600000,
    0.02601606,
  ],
  Array [
    1604188800000,
    0.02612575,
  ],
  Array [
    1604448000000,
    0.02514066,
  ],
  Array [
    1604707200000,
    0.02945723,
  ],
  Array [
    1604966400000,
    0.02761045,
  ],
  Array [
    1605225600000,
    0.02765661,
  ],
  Array [
    1605484800000,
    0.02843925,
  ],
  Array [
    1605744000000,
    0.03089734,
  ],
  Array [
    1606003200000,
    0.03378528,
  ],
  Array [
    1606262400000,
    0.0356698,
  ],
  Array [
    1606521600000,
    0.03466291,
  ],
  Array [
    1606780800000,
    0.04105239,
  ],
  Array [
    1607040000000,
    0.04671706,
  ],
  Array [
    1607299200000,
    0.0492354,
  ],
  Array [
    1607558400000,
    0.04308045,
  ],
  Array [
    1607817600000,
    0.04485471,
  ],
  Array [
    1608076800000,
    0.04779049,
  ],
  Array [
    1608336000000,
    0.05919994,
  ],
  Array [
    1608554600198,
    0.06553323,
  ],
]
`;
